flag_impl.go

```diff
@@ -162,13 +153,29 @@ func (f *FlagBase[T, C, V]) Apply(set *flag.FlagSet) error {
 		}
 	}
 
-	vw := &valueWrapper{
-		value:    f.value,
-		onlyOnce: f.OnlyOnce,
+	isBool := false
+	if b, ok := f.value.(boolFlag); ok && b.IsBoolFlag() {
+		isBool = true
 	}
 
 	for _, name := range f.Names() {
-		set.Var(vw, name, f.Usage)
+		set.Var(fnValue{
+			fn: func(val string) error {
+				if f.count == 1 && f.OnlyOnce {
+					return fmt.Errorf("cant duplicate this flag")
+				}
+				f.count++
+				if err := f.value.Set(val); err != nil {
+					return err
+				}
+				if f.Validator != nil {
+					return f.Validator(f.value.Get().(T))
```

Comment:
I know this isn't the first place we have an unchecked type coercion, but for some reason this one makes me really nervous 😅 Is there an alternative that would allow us to bubble up an error instead of panicking?

Reply:
I understand. For now lets just roll with this. Maybe we can enforce a stronger type check in all the other ctx.Type(str) functions as well along with this. It is mostly safe if you use internal libraries. For user defined types yes it is problematic. Lets revisit later